### DISCUSSION TITLE
Is the copyright holder of this software really Amazon ?

### DISCUSSION THREAD

**Discussion by DerAndere1** [feilipu/Arduino_FreeRTOS_Library#85] [Q&A] [answered]
Dear feilipu,

In the LICENSE file for the Arduino_FreeRTOS_library, the Copyright notice is as follows:

https://github.com/feilipu/Arduino_FreeRTOS_Library/blob/329328e050eed1b59da013af2ca6e697b0be601d/LICENSE#L2

I just wanted to mention it, in case it is a copy/paste mistake.

**Comment by feilipu** [comment, marked as answer]
Thanks for asking.

Yes. Generally, the copywrite holder is Amazon, since they purchased the rights from FreeRTOS and re-released the code under MIT licence. The base code is kept, to as great an extent as practical, identical to the code in the main [FreeRTOS Kernel](https://github.com/FreeRTOS/FreeRTOS-Kernel) repository, but redacted into just the files relevant here.

Also, most of my own work here has been [accepted into the mainline FreeRTOS code as the ATmegaxxx port](https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/48), which also implies providing the copyright to Amazon (under MIT terms).

There are a few files that retain copyright by me, and they're noted as such in the header of the file.
